### PR TITLE
[chore] Add an automation to skip release if there no changes

### PR DESCRIPTION
**Pull request by dmitryax**
As discussed at one of the recent SIGs, we don't want to release module sets with no changes.

This change updates the release process to skip releasing the stable set of modules if there are no changes since the last version.

Generated PR sample: https://github.com/dmitryax/opentelemetry-collector/pull/3

### CODE REVIEW

**Comment by dmitryax** [comment]
The failing test is unrelated, fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/7561

File: .github/workflows/scripts/release-create-tracking-issue.sh [hunk 1]
```diff
@@ -1,19 +1,24 @@
 #!/bin/bash -ex
 
-EXISTING_ISSUE=$( gh issue list --search "Release v${CANDIDATE_BETA}/v${CANDIDATE_STABLE}" --json url --jq '.[].url' --repo "${REPO}" )
+RELEASE_VERSION=v${CANDIDATE_STABLE}/v${CANDIDATE_BETA}
+if [ "${CANDIDATE_STABLE}" == "" ]; then
+    RELEASE_VERSION=v${CURRENT_STABLE}/v${CANDIDATE_BETA}
+elif [ "${CANDIDATE_BETA}" == "" ]; then
+    RELEASE_VERSION=v${CANDIDATE_STABLE}/v${CURRENT_BETA}
+fi
```
Comment (codeboten) [inline .github/workflows/scripts/release-create-tracking-issue.sh:8]:
```suggestion
STABLE="${CANDIDATE_STABLE}"
if [ "${STABLE}" == "" ]; then
    STABLE="${CURRENT_STABLE}"
fi

BETA="${CANDIDATE_BETA}"
if [ "${BETA}" == "" ]; then
    BETA="${CURRENT_BETA}"
fi

RELEASE_VERSION=v${STABLE}/v${BETA}
```

File: .github/workflows/scripts/release-prepare-release.sh [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
+    echo "One of CANDIDATE_STABLE or CANDIDATE_BETA must be set"
+    exit 1
+fi
+
+RELEASE_VERSION=v${CANDIDATE_STABLE}/v${CANDIDATE_BETA}
+if [ "${CANDIDATE_STABLE}" == "" ]; then
+    RELEASE_VERSION=v${CURRENT_STABLE}/v${CANDIDATE_BETA}
+elif [ "${CANDIDATE_BETA}" == "" ]; then
+    RELEASE_VERSION=v${CANDIDATE_STABLE}/v${CURRENT_BETA}
+fi
```
Comment (codeboten) [inline .github/workflows/scripts/release-prepare-release.sh:13]:
```suggestion
STABLE="${CANDIDATE_STABLE}"
if [ "${STABLE}" == "" ]; then
    STABLE="${CURRENT_STABLE}"
fi

BETA="${CANDIDATE_BETA}"
if [ "${BETA}" == "" ]; then
    BETA="${CURRENT_BETA}"
fi

RELEASE_VERSION=v${STABLE}/v${BETA}
```

**Review by codeboten — COMMENTED** [review]
Thanks for taking this on, just a couple of suggestions

File: .github/workflows/scripts/release-create-tracking-issue.sh (hunk 1 above)
Comment (dmitryax) [reply .github/workflows/scripts/release-create-tracking-issue.sh]: Thanks. applied

**Review by codeboten — COMMENTED** [review]
What happens with `make push-tags` when there's no change? will it fail when trying to create a tag that already exists?

**Comment by dmitryax** [comment]
> What happens with make push-tags when there's no change? will it fail when trying to create a tag that already exists?

Yes, it'll just fail to push existing tags. I updated the release section for `make push-tags` to make it clear when to run them

**Review by codeboten — APPROVED** [review]

**Pull request merged**